origin-bridge/src/utils/index.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const Web3 = require('web3')
+const web3 = new Web3(process.env.PROVIDER_URL || 'http://localhost:8545')
+const dictionary = require('./dictionary')
+
+function generateAirbnbCode(ethAddress, userId) {
+  const hashCode = web3.utils.sha3(ethAddress + userId).substr(-7)
+  return Array.prototype.map
+    .call(hashCode, i => dictionary[i.charCodeAt(0)])
+    .join(' ')
+}
+
+function generateSixDigitCode() {
+  return Math.floor(100000 + Math.random() * 900000)
+}
+
+function getAbsoluteUrl(relativeUrl, dappRedirectUrl = null) {
+  const protocol = process.env.HTTPS ? 'https' : 'http'
+  const host = process.env.HOST ? process.env.HOST : 'localhost:5000'
+  let url = `${protocol}://${host}${relativeUrl}`
+  if (dappRedirectUrl) {
+    url += '?dappRedirectUrl=' + dappRedirectUrl
+  }
+  return url
+}
+
+function mapObjectToQueryParams(obj) {
+  return Object.keys(obj)
+    .map(key => key + '=' + obj[key])
```

Comment:
keys and values should probably be URL encoded ? Perhaps using encodeURIComponent